### ISSUE TITLE
[3.9] Strange logic of the Terms of Service clause

### ISSUE THREAD

**Issue by Kostelano** [joomla/joomla-cms#28537]
### Steps to reproduce the issue
1. Use the Joomla 3.9.16,
2. Turn on the option Register users on your site,
3. Go to Extensions -> Plugins -> User - Profile, select a value for Terms of Service -> Required, but DO NOT select the article with the conditions (below),
4. Go to the front-end and try to register, but DO NOT check the Terms of Service checkbox.

### Expected result
Error because I did NOT check the Terms of Service checkbox.

### Actual result
I registered on the site without any problems, but did NOT check the Terms of Service item. Please note that the hint tells us the following:

![Screenshot_1](https://user-images.githubusercontent.com/8440661/78173967-3484f580-7461-11ea-8234-d9fe759a39cc.png)

### System information (as much as possible)
Joomla 3.9.16
PHP 7.3.2

### Additional comments
If you select an article with conditions in the plugin settings, then everything works correctly - the field is required and registration is not possible.

You need to either remove part of the tooltip from the language constant:

`PLG_USER_PROFILE_FIELD_TOS_DESC_SITE="Please read the Terms of Service. You will not be able to register if you do not agree with them."`

Or, check for an article with conditions and, possibly, do not display the field even if the value is selected, which is mandatory.

In short, the logic is strange and needs some refinement :) Thanks.

**Comment by infograf768** [comment]
Confirmed.
Looks like we should make the selection or creation of the Tos article mandatory when Tos is required. Also, the field should display only when Tos is required with a showon.

**Comment by brianteeman** [comment]
Thank you for raising this issue.

Joomla 3 is now in security only mode with no further bug fixes or new features.

As this issue still exists in Joomla 4 I have created a pull request for testing #38619
cc @zero-24
